### DISCUSSION TITLE
configuring lspi peripheral to not use built-in chip select

### DISCUSSION THREAD

**Discussion by lehmanpm** [mciantyre/teensy4-rs#167] [Q&A] [answered]
I'm fairly new to rust, and am trying to convert the firmware I wrote for a board containing a Teensy 4.0 with multiple instances of a max31855 spi IC.  I'm in the process of trying to base my code off the rtic lspi example to setup lspi4 but since I have multiples of these ICs all on the same spi bus, I want to use a variety of gpios to act as the chip selects instead of just assigning on at initialization.  

So, is there an easy way to specify None for the PCS0 pin? or some other method to not specify a Chip select when trying to use the driver?

From trying to dig into the teensy4-rs lpspi.rs that I might be able to somehow us the init implementation on line 465 if I could change CFGR1 to set the AUTOPCS to disabled, but I'll be honest, that appears a little beyond me at the moment, so anything that could help would be greatly appreciated.

**Comment by mciantyre** [comment]
https://github.com/mciantyre/teensy4-rs/discussions/131 talks about how folks can use **software**-managed chip selects with today's driver. Make sure you're on imxrt-hal 0.5.5 or newer, since that's where we start supporting this use-case. See https://github.com/mciantyre/teensy4-rs/discussions/159 for more details.

Although it doesn't sound like you need it, note that https://github.com/imxrt-rs/imxrt-hal/issues/131 is tracking **hardware**-managed chip selects.

**Comment by lehmanpm** [comment, marked as answer]
Thank you for the quick response. Can’t understand how I missed this during
my search, but that is what I’m looking for..

Much appreciated.
